Manager/Manager.php

```diff
@@ -78,6 +89,32 @@ public function save($entity, array $options = array())
      */
     public function remove($entity)
     {
-        $this->remover->remove($entity);
+        if (!$this->isLinkedToProduct($entity)) {
+            $this->remover->remove($entity);
+        }
+    }
+
+    /**
+     * Check if the entity is linked to one or more products
+     *
+     * @param ReferenceDataInterface|object $entity
+     *
+     * @return bool
+     */
+    protected function isLinkedToProduct($entity)
+    {
+        $metadata = $this->em->getClassMetadata(Attribute::class);
+        $repository = new AttributeRepository($this->em, $metadata);
+        $attributesCodes = $repository
+            ->findReferenceDataAttributeCodesByEntityName($entity->getCode());
+
+        $pqb = $this->productQueryBuilderFactory->create();
+        foreach ($attributesCodes as $attributeCode) {
+            $pqb->addFilter($attributeCode, Operators::IN_LIST, [$entity->getCode()]);
```

Comment:
Adding many filter will add a `AND` clause.
You want to add an `OR` clause which is not possible with the current PQB.
So you need to count per attribute ;)

Reply:
:+1: